packages/gatsby/src/db/types.ts

```diff
@@ -0,0 +1,18 @@
+import { GraphQLType } from "graphql"
+import { Node } from "../../index"
+
+export interface INodeStore {
+  getNodes: () => Node[]
+  getNode: (id: string) => Node | undefined
+  getNodesByType: (type: string) => Node[]
+  getTypes: () => string[]
+  hasNodeChanged: (id: string, digest: string) => boolean
+  getNodeAndSavePathDependency: (id: string, path: string) => Node | undefined
+  runQuery: (args: {
+    gqlType: GraphQLType
+    queryArgs: Record<string, any>
+    firstOnly: boolean
+    resolvedFields: Record<string, any>
+    nodeTypeNames: string[]
+  }) => any | undefined
+}
```

Comment:
I took this definition from what I think is the Flow interface in `nodes.js`. I wasn't sure if mixing the Node from several levels up with this `INodeStore` was the way to go or not.

`runQuery` returns an `any` for now. I couldn't convince myself of a definitive type for it, but it's probably `Promise<Node[]>`, right?